### PR TITLE
refactor: deduplicate block downloading and incoming requests handling

### PR DESCRIPTION
**Pull request by MegaRedHand**
This PR removes duplicate code in `BlockDownloader` and `IncomingRequests.Handler`, simplifying it in the process, and making it easier to add the blob-related req/resp callbacks in the future. This should also make it easier to test since encoding/decoding is now contained in a few functions, that can be exposed for testing.

Also, it adds testing for the new module.

### CODE REVIEW

File: lib/lambda_ethereum_consensus/p2p/block_downloader.ex [hunk 1]
```diff
@@ -15,100 +17,68 @@ defmodule LambdaEthereumConsensus.P2P.BlockDownloader do
...
-
-    with {:ok, response_chunk} <-
-           Libp2pPort.send_request(
-             peer_id,
-             @blocks_by_root_protocol_id,
-             size_header <> compressed_payload
-           ),
-         {:ok, chunks} <- parse_response(response_chunk),
-         {:ok, blocks} <- decode_chunks(chunks) do
+    schema = {:list, TypeAliases.root(), ChainSpec.get("MAX_REQUEST_BLOCKS")}
```
Comment (Arkenan) [inline lib/lambda_ethereum_consensus/p2p/block_downloader.ex:76]: Maybe we should have a type alias of this? e.g. `TypeAliases.block_root_list`? We are using this exact same thing in  `BeaconBlockByRootRequestContainer`. If we can't use that container directly, we should at least alias this list.

**Review by Arkenan — APPROVED** [review]
Great refactor! Thanks for taking care of this.

File: lib/lambda_ethereum_consensus/p2p/req_resp.ex [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  def decode_response(response_chunk) do
+    with {:ok, chunks} <- split_response(response_chunk) do
+      # TODO: handle errors
+      chunks
+      |> Enum.map(&decode_request(&1, SignedBeaconBlock))
+      |> Enum.map(fn
+        {:ok, block} -> block
+        {:error, _reason} -> nil
+      end)
+      |> Enum.filter(&(&1 != nil))
```
[the remark this answers is not in the document]
Comment (MegaRedHand) [reply lib/lambda_ethereum_consensus/p2p/req_resp.ex:66]: Thanks for the advice!

File: lib/types/p2p/beacon_blocks_ by_root_request.ex [hunk 3]
```diff
@@ -3,6 +3,7 @@ defmodule Types.BeaconBlocksByRootRequest do
   Struct definition for `BeaconBlocksByRootRequest`.
   Related definitions in `native/ssz_nif/src/types/`.
   """
```
[the remark this answers is not in the document]
Comment (MegaRedHand) [reply lib/types/p2p/beacon_blocks_ by_root_request.ex:5]: Of course. Good catch!

File: lib/lambda_ethereum_consensus/p2p/block_downloader.ex (hunk 1 above)
Comment (MegaRedHand) [reply lib/lambda_ethereum_consensus/p2p/block_downloader.ex]: I changed it so it now uses the struct instead of the raw schema.

**Pull request merged**